src/use-cases/createUseCase.ts

```diff
@@ -0,0 +1,47 @@
+import { hash } from 'bcryptjs'
+import { UserRepository } from '../repositories/user-repository'
+import { UserAlreadyExistsError } from './errors/user-already-exists-error'
+import { User } from '@prisma/client'
+
+interface CreateUseCaseRequest {
+  name: string
+  surname: string
+  email: string
+  password: string
+}
+
+interface CreateUseCaseResponse {
+  user: User
+}
+
+export class CreateUseCase {
+  constructor(private usersRepository: UserRepository) {}
+
+  async execute({ 
+    name, 
+    surname, 
+    email, 
+    password, 
+  }: CreateUseCaseRequest): Promise<CreateUseCaseResponse> {
+    const password_hash = await hash(password, 6)
```

Comment:
Minor: This line could be moved after you verify if the user already exists.

There's no need to hash the password before checking if the user exists.